src/db/api/validator/interface.ts

```diff
@@ -0,0 +1,30 @@
+/**
+ * @module db/api/validator
+ */
+
+import {Attestation, BeaconBlock, ValidatorIndex} from "../../../types";
+
+export interface IValidatorDB {
+
+  /**
+   * Obtains last proposed beacon block
+   * by validator with given index
+   */
+  getBlock(index: ValidatorIndex): Promise<BeaconBlock>;
+
+  /**
+   * Stores beacon block proposed by validator with given index
+   */
+  setBlock(index: ValidatorIndex, block: BeaconBlock): Promise<void>;
+
+  /**
+   * Obtains last attestation proposed by validator with given index
+   */
+  getAttestation(index: ValidatorIndex): Promise<Attestation>;
```

Comment:
I think we may want to be able to fetch attestations by some sort of index?

Reply:
Damm, I will check, it seemed that only last proposed attestation can be in conflict.